spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/PropertiesResultSetExtractor.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.config.server.environment;
+
+import java.util.Map;
+
+import org.springframework.jdbc.core.ResultSetExtractor;
+
+/**
+ * @author Chintan Radia
+ */
+public interface PropertiesResultSetExtractor extends ResultSetExtractor<Map<String, ?>> {
```

Comment:
Not sure I understand the need for this interface

Reply:
was trying to expose an interface which a developer can implement to provide a custom PropertyResultSetExtractor implementation.
Without this, we have to modify the existing class as below.
```java
public static class PropertiesResultSetExtractor implements ResultSetExtractor<Map<String, Object>> {

		@Override
		public Map<String, Object> extractData(ResultSet rs) throws SQLException, DataAccessException {
			Map<String, Object> map = new LinkedHashMap<>();
			while (rs.next()) {
				map.put(rs.getString(1), rs.getString(2));
			}
			return map;
		}
	}
```

and if anyone wants to provide customer implementation, they have to create a child class and provide its bean.
and using AutoConfiguration class we can control If we want to use `PropertiesResultSetExtractor` or `CustomPropertiesResultSetExtractor`

```java
public static class CustomResultSetExtractor extends JdbcEnvironmentRepository.PropertiesResultSetExtractor {
@Override
public Map<String, Object> extractData(ResultSet rs) throws SQLException, DataAccessException {
	Map<String, Object> map = new LinkedHashMap<>();
	while (rs.next()) {
		map.put(rs.getString(1), rs.getInt(2));
	}
	return map;
}
}


@Bean
@ConditionalOnMissingBean(value = JdbcEnvironmentRepository.PropertiesResultSetExtractor.class)
public JdbcEnvironmentRepository.PropertiesResultSetExtractor propertiesResultSetExtractor() {
	return new JdbcEnvironmentRepository.PropertiesResultSetExtractor();
}
```

Is this the better approach?

Reply:
You could just make `PropertiesResultSetExctractor` a top level class, that might be easiest, then users can just extend that.  Then use the `@ConditionalOnMissingBean` approach.